src/test/java/org/broadinstitute/hellbender/tools/GatherVcfsCloudIntegrationTest.java

```diff
@@ -31,47 +35,48 @@
-                                            getTestFile("cloud-inputs.args"));
+                new File(largeFileTestDir, "1000G.phase3.broad.withGenotypes.chr20.10100000.vcf"),
+                getTestFile("cloud-inputs.args"));
     }
 
     @Test(dataProvider = "getGatherTypes")
-    public void testLocalGathers(final GatherVcfsCloud.GatherType gatherType){
+    public void testLocalGathers(final GatherVcfsCloud.GatherType gatherType) throws IOException {
         assertGatherProducesCorrectVariants(gatherType,
-                                            getTestFile("gzipped.vcf.gz"),
-                                            getTestFile("bgzipped_shards.args"));
+                getTestFile("gzipped.vcf.gz"),
+                getTestFile("bgzipped_shards.args"));
     }
 
-    private void assertGatherProducesCorrectVariants(GatherVcfsCloud.GatherType gatherType, File expected, File inputs) {
+    private void assertGatherProducesCorrectVariants(GatherVcfsCloud.GatherType gatherType, File expected, File inputs) throws IOException {
         assertGatherProducesCorrectVariants(gatherType, expected, Collections.singletonList(inputs));
     }
 
-    private void assertGatherProducesCorrectVariants(GatherVcfsCloud.GatherType gatherType, File expected, List<File> inputs) {
+    @SuppressWarnings({"unchecked"})
+    private void assertGatherProducesCorrectVariants(GatherVcfsCloud.GatherType gatherType, File expected, List<File> inputs) throws IOException {
         final File output = createTempFile("gathered", ".vcf.gz");
         final ArgumentsBuilder args = new ArgumentsBuilder();
         inputs.forEach(args::addInput);
         args.addOutput(output)
                 .addArgument(GatherVcfsCloud.GATHER_TYPE_LONG_NAME, gatherType.toString());
         runCommandLine(args);
 
-        try (  final FeatureDataSource<VariantContext> actualReader = new FeatureDataSource<>(output);
-               final FeatureDataSource<VariantContext> expectedReader = new FeatureDataSource<>(expected);){
+        try (final AbstractFeatureReader<VariantContext, ?> actualReader = AbstractFeatureReader.getFeatureReader(output.getAbsolutePath(), null, new VCFCodec(), false, Function.identity(), Function.identity());
+             final AbstractFeatureReader<VariantContext, ?> expectedReader = AbstractFeatureReader.getFeatureReader(expected.getAbsolutePath(), null, new VCFCodec(), false, Function.identity(), Function.identity())) {
 
-            final List<VariantContext> actualVariants = Lists.newArrayList(actualReader);
-            final List<VariantContext> expectedVariants = Lists.newArrayList(expectedReader);
+            final List<VariantContext> actualVariants = Lists.newArrayList((Iterable<? extends VariantContext>) actualReader);
+            final List<VariantContext> expectedVariants = Lists.newArrayList((Iterable<? extends VariantContext>) actualReader);
```

Comment:
Looks like you've accidentally clobbered this test -- now you're comparing the actual output against itself.

Also, given that `AbstractFeatureReader` does not implement `Iterable`, you should instead call the overload that takes an `Iterator` (by passing in `reader.iterator()`), and remove the typecast.